### PR TITLE
Update Unit test for FindMaxTest, FindMinRecursionTest, ReverseNumberTest, UpperTest

### PR DESCRIPTION
**Pull request by ThieuDuong0111**
update Unit Test

- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized it.
- [ ] All filenames are in PascalCase.
- [ ] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [ ] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/test/java/com/thealgorithms/maths/FindMaxTest.java [hunk 1]
```diff
@@ -25,4 +25,28 @@ private static Stream<Arguments> inputStream() {
...
+    public void testFindMaxWithOneElement() {
+        int[] array = {42};
+        Assertions.assertEquals(42, FindMax.findMax(array));
+    }
+
+    @Test
+    public void testFindMaxWithEmptyArray() {
+        int[] emptyArray = {};
+        assertThrows(IllegalArgumentException.class, () -> { FindMax.findMax(emptyArray); });
+    }
```
Comment (vil02) [inline src/test/java/com/thealgorithms/maths/FindMaxTest.java:51]: This is already tested in `testFindMaxThrowsExceptionForEmptyInput`.

File: src/test/java/com/thealgorithms/maths/ReverseNumberTest.java [hunk 2]
```diff
@@ -25,5 +25,15 @@ public void testReverseNumber() {
...
+
+    @Test
+    public void testReverseNumberWithSingleDigit() {
+        assertEquals(5, ReverseNumber.reverseNumber(5));
+    }
+
+    @Test
+    public void testReverseNumberWithNegativeNumber() {
+        assertThrows(IllegalArgumentException.class, () -> { ReverseNumber.reverseNumber(-123); });
     }
```
Comment (vil02) [inline src/test/java/com/thealgorithms/maths/ReverseNumberTest.java:38]: This is already covered in `testReverseNumberThrowsExceptionForNegativeInput`.

File: src/test/java/com/thealgorithms/searches/PerfectBinarySearchTest.java [hunk 3]
```diff
@@ -1,3 +1,5 @@
+package com.thealgorithms.searches;
+
```
Comment (vil02) [inline src/test/java/com/thealgorithms/searches/PerfectBinarySearchTest.java:2]: Why is this needed here?

File: src/test/java/com/thealgorithms/strings/UpperTest.java [hunk 4]
```diff
@@ -15,4 +15,27 @@ public void toUpperCase() {
...
+    
+    @Test
+    public void testToUpperCase() {
+        String[] strings = {"ABC", "ABC123", "abcABC", "abc123ABC"};
+        String[] expected = {"ABC", "ABC123", "ABCABC", "ABC123ABC"};
+
+        for (int i = 0; i < strings.length; i++) {
+            String result = Upper.toUpperCase(strings[i]);
+            assertEquals(expected[i], result);
+        }
```
Comment (vil02) [inline src/test/java/com/thealgorithms/strings/UpperTest.java:27]: You could express these tests as _parametrized_ tests (cf. `FindMaxTest.java`).

**Review by vil02 — COMMENTED** [review]
Most of the new tests cases do not add any value (some of them are even duplicates of existing ones).

@siriak please have a look.

**Review by siriak — CHANGES_REQUESTED** [review]
Please remove duplicated tests and apply the suggested changes. And please also fix failing linter job.
